core/src/test/java/kafka/server/integration/EligibleLeaderReplicasIntegrationTest.java

```diff
@@ -0,0 +1,450 @@
+            waitForIsrAndElr((isrSize, elrSize) -> {
+                return isrSize == 4 && elrSize == 0;
+            });
+
+            Consumer finalConsumer = consumer;
+            kafka.utils.TestUtils.waitUntilTrue(
+                () -> {
+                    try {
+                        ConsumerRecords record = finalConsumer.poll(Duration.ofMillis(100L));
+                        return record.count() == 1;
+                    } catch (Exception e) {
+                        return false;
+                    }
+                },
+                () -> "fail to consume messages",
+                org.apache.kafka.test.TestUtils.DEFAULT_MAX_WAIT_MS, 100L
+            );
+        } finally {
+            restartDeadBrokers(false);
+            if (consumer != null) consumer.close();
+            if (producer != null) producer.close();
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"kraft"})
+    public void testElrMemberCanBeElected(String quorum) throws ExecutionException, InterruptedException {
+        adminClient.createTopics(
+            Collections.singletonList(new NewTopic(testTopicName, 1, (short) 4))).all().get();
+        TestUtils.waitForPartitionMetadata(brokers(), testTopicName, 0, 1000);
+
+        ConfigResource configResource = new ConfigResource(ConfigResource.Type.TOPIC, testTopicName);
+        Collection<AlterConfigOp> ops = new ArrayList<>();
+        ops.add(new AlterConfigOp(new ConfigEntry(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG, "3"), AlterConfigOp.OpType.SET));
+        Map<ConfigResource, Collection<AlterConfigOp>> configOps = Collections.singletonMap(configResource, ops);
+        // alter configs on target cluster
+        adminClient.incrementalAlterConfigs(configOps).all().get();
+
+        try {
+            // check which partition is on broker 0 which we'll kill
+            TopicDescription testTopicDescription = adminClient.describeTopics(Collections.singletonList(testTopicName))
+                .allTopicNames().get().get(testTopicName);
+            TopicPartitionInfo topicPartitionInfo = testTopicDescription.partitions().get(0);
+            List<Node> initialReplicas = topicPartitionInfo.replicas();
+            assertEquals(4, topicPartitionInfo.isr().size());
+            assertEquals(0, topicPartitionInfo.elr().size());
+            assertEquals(0, topicPartitionInfo.lastKnownElr().size());
+
+            killBroker(initialReplicas.get(0).id());
+            killBroker(initialReplicas.get(1).id());
+            killBroker(initialReplicas.get(2).id());
+
+            waitForIsrAndElr((isrSize, elrSize) -> {
+                return isrSize == 1 && elrSize == 2;
+            });
+
+            topicPartitionInfo = adminClient.describeTopics(Collections.singletonList(testTopicName))
+                .allTopicNames().get().get(testTopicName).partitions().get(0);
+            assertEquals(2, topicPartitionInfo.elr().size());
```

Comment:
This seems unnecessary since it's covered by the `waitForIsrAndElr` call earlier?